internal/notifier/gitea.go

```diff
@@ -52,11 +52,11 @@ func NewGitea(providerUID string, addr string, token string, certPool *x509.Cert
 
 	host, id, err := parseGitAddress(addr)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed parsing Git URL: %w", err)
```

Comment:
Fixed the missing error returned from `parseGitAddress`.

> For the url.Parse() error below, instead of adding the context to all the git related notifiers for the same, maybe we can parse the host in parseGitAddress() itself and add the appropriate context just there.

Not all notifiers parse the host in the same way so it can't be added to `parseGitAddress`.

Reply:
> Irrespective of that, parseGitAddress() can guarantee that the host value it returns is valid. The consumer of the result of parseGitAddress() don't have to again parse to make sure they have a valid host.

This would change the API, though and that's why I didn't want to do that as part of this PR. It could lead to a change in behaviour for those notifiers that don't care about the host part.

> Also, it looks like in the above code, there will be double context with redundant message about failed parsing URL.

Yes, I'd happily take suggestions to improve the wording of the errors.